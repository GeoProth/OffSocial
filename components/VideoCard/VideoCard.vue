<template>
  <div class="video-card">
    <video controls class="video" type="video/mp4">
      <source :src="url">
    </video>
    <h3>{{ name }}</h3>
    <DownloadButton :url="url" />
  </div>
</template>
<script>
export default {
  name: "VideoCard",
  props: {
    url: String,
    name: String
  }
};
</script>
<style lang="stylus" scoped>
.video-card {
  height: fit-content;
  width: fit-content;
  margin-top: 2rem;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 2rem;
  align-self: center;
  box-shadow: -2px 2px 12px -1px;
  border-radius: 4px;
  text-align: center;

  &:hover {
    box-shadow: -2px 7px 12px -1px;
    transform: scale(1.025, 1.025);
  }

  .video {
    width: 100%;
    border-radius: 4px;
    outline: none;
  }
}
</style>
