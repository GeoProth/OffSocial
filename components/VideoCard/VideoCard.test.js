import VideoCard from "./VideoCard";
describe("VideoCard", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(VideoCard);
  });
  test("renders", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
