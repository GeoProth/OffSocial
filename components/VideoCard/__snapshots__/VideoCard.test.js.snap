// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCard renders 1`] = `
<div class="video-card"><video controls="controls" type="video/mp4" class="video">
    <source></video>
  <h3></h3>
  <downloadbutton-stub text="Download"></downloadbutton-stub>
</div>
`;
